File: Java/VK_info/app/src/main/java/com/example/android/vk_info/MainActivity.java
package com.example.android.vk_info;

import androidx.appcompat.app.AppCompatActivity;

import android.os.AsyncTask;
import android.os.Bundle;
import android.view.View;
import android.widget.EditText;
import android.widget.ProgressBar;
import android.widget.TextView;

import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;

import java.io.IOException;
import java.net.URL;

import static com.example.android.vk_info.utils.NetworkUtils.generalURL;
import static com.example.android.vk_info.utils.NetworkUtils.getResponseFromURL;

public class MainActivity extends AppCompatActivity {
    private EditText etNumberID;
    private TextView tvFoundText;
    private TextView tvErrorMassage;
    private ProgressBar pbLoadIndicator;

    private void showTvFoundText() {
        tvErrorMassage.setVisibility(View.INVISIBLE);
        tvFoundText.setVisibility(View.VISIBLE);
    }

    private void showTvErrorMassage() {
        tvFoundText.setVisibility(View.INVISIBLE);
        tvErrorMassage.setVisibility(View.VISIBLE);

    }

    // работа в другом (не в главном) потоке
    class VKQueryTask extends AsyncTask<URL, Void, String> {
        //до
        @Override
        protected void onPreExecute() {
            pbLoadIndicator.setVisibility(View.VISIBLE);
        }

        //работа в другом потоке
        @Override
        protected String doInBackground(URL... urls) {
            String response = null;
            try {
                response = getResponseFromURL(urls[0]);
            } catch (IOException e) {
                e.printStackTrace();
            }
            return response;
        }

        //после
        @Override
        protected void onPostExecute(String response) {
            String firstName = null;
            String lastName = null;
            if (response != null && !response.equals("")) {
                try {
                    JSONObject jsonResponse = new JSONObject(response);
                    JSONArray jsonArray = jsonResponse.getJSONArray("response");
                    JSONObject userInfo = jsonArray.getJSONObject(0);

                    firstName = userInfo.getString("first_name");
                    lastName = userInfo.getString("last_name");
                } catch (JSONException e) {
                    e.printStackTrace();
                }
                String result = "Имя: " + firstName + "\n" + "Фамилия: " + lastName;
                tvFoundText.setText(result);

                showTvFoundText();
            } else
                showTvErrorMassage();

            pbLoadIndicator.setVisibility(View.INVISIBLE);
        }
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        etNumberID = findViewById(R.id.et_numberID);
        tvFoundText = findViewById(R.id.tv_foundText);
        tvErrorMassage = findViewById(R.id.tv_error_massage);
        pbLoadIndicator = findViewById(R.id.pb_load_indicator);

    }

    //нажатие кнопки
    public void click_button_find(View view) {
        URL url = generalURL(etNumberID.getText().toString());
        new VKQueryTask().execute(url);
    }
}

File: Kotlin/DownloadWebKontent/app/src/main/java/com/example/android/downloadwebkontent/MainActivity.kt
package com.example.android.downloadwebkontent

import android.os.AsyncTask
import android.os.Bundle
import android.util.Log
import androidx.appcompat.app.AppCompatActivity
import java.io.BufferedReader
import java.io.IOException
import java.io.InputStreamReader
import java.net.HttpURLConnection
import java.net.URL
import kotlin.text.StringBuilder as StringBuilder


class MainActivity : AppCompatActivity() {

    private val mailRu = "https://mail.ru/"

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val task = DownloadTask()

        val result = task.execute(mailRu).get()
        Log.i("URL", result)

    }

    class DownloadTask : AsyncTask<String, Unit, String>() {

        override fun doInBackground(vararg params: String?): String {
            val result = StringBuilder()
            var url: URL? = null
            var urlConnection: HttpURLConnection? = null
            try {
                url = URL(params[0])
                urlConnection = url.openConnection() as HttpURLConnection
                val reader = InputStreamReader(urlConnection.inputStream)
                val bufferReader = BufferedReader(reader)
                var line = bufferReader.readLine()
                while (line != null) {
                    result.append(line)
                    line = bufferReader.readLine()
                }
            } catch (e: IOException) {
                e.printStackTrace()
            } finally {
                urlConnection?.disconnect()

                return result.toString()
            }
        }
    }
}
